homeassistant/helpers/template.py

```diff
@@ -329,10 +330,22 @@ def async_render(self, variables: TemplateVarsType = None, **kwargs: Any) -> str
             kwargs.update(variables)
 
         try:
-            return compiled.render(kwargs).strip()
+            render_result = compiled.render(kwargs)
         except jinja2.TemplateError as err:
             raise TemplateError(err) from err
 
+        if not self.hass.config.legacy_templates:
+            try:
+                result = literal_eval(render_result)
+
+                # If the literal_eval result is a string, use the original render
+                if not isinstance(result, str):
+                    return result
```

Comment:
Why do we do this? What would be the use case for/against this case? Let's add it as a comment here too maybe?

Reply:
I've extended the comment on why this is.

`literal_eval` on a string (which results in a string) can result in different contents. Especially when working with quotes. As this is only a concern for strings, simply not using the evaluated result, if the evaluated result is a string, resolves it.

Reply:
You can actually test/replicate this, by uncommenting these 2 lines and run the `command_line` integration tests.